### PR TITLE
MLPAB-2125: Use transactions when creating attorneys

### PR DESCRIPTION
**Pull request by acsauk**
# Purpose

Fixes MLPAB-2125

### CODE REVIEW

File: internal/app/attorney_store.go [hunk 1]
```diff
@@ -29,24 +29,37 @@ func (s *attorneyStore) Create(ctx context.Context, lpaOwnerKey dynamo.LpaOwnerK
...
 		ActorType: actor.TypeAttorney,
 		UpdatedAt: s.now(),
-	}); err != nil {
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	transaction := dynamo.NewTransaction().
+		Put(marshalledAttorney).
```
Comment (hawx) [inline internal/app/attorney_store.go:57]: I like this, my only suggestion would be to push the `MarhsalMap` calls in to `Put`. You could use a pattern like https://go.dev/blog/errors-are-values to hold onto the errors, then return in `WriteTransaction`

File: internal/page/fixtures/attorney.go [hunk 2]
```diff
@@ -384,3 +481,27 @@ func Attorney(
...
+	if isReplacement {
+		attorneyType = actor.TypeReplacementAttorney
+	}
+
+	err := shareCodeStore.Put(ctx, attorneyType, shareCode, shareCodeData)
+	if err != nil {
+		return nil, err
+	}
+
+	return attorneyStore.Create(ctx, shareCodeData, email)
```
Comment (hawx) [inline internal/page/fixtures/attorney.go:506]: This is fine, it's the price we pay. I'd rather have the actual code and fixtures a mess. It's only for setting up test data so we shouldn't worry too much about it

File: internal/app/attorney_store.go (hunk 1 above)
Comment (acsauk) [reply internal/app/attorney_store.go:57]: Ah cool, that's precisely what I was trying to avoid so thanks for the link 👍

Comment (acsauk) [reply internal/app/attorney_store.go:57]: (and this will make testing so much easier)

**Review by hawx — APPROVED** [review]

**Pull request merged**